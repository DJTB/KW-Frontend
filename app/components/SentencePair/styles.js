import styled from 'styled-components';
import { gutter } from 'shared/styles/layout';

export const Wrapper = styled.div`
  ${gutter()}
  display: flex;
  flex-flow: column nowrap;
  align-items: inherit;
  max-width: 42rem;
`;
