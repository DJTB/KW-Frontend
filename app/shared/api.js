import urljoin from 'url-join';
import { get, put, post, patch, del } from 'utils/request';

const BASE_URL = '//localhost:8000'; // FIXME: change for production
const API_BASE = urljoin(BASE_URL, 'api', 'v1');

//-----------------------------------------------------------------------------
//  AUTHORIZATION
//-----------------------------------------------------------------------------
const authUrl = urljoin(API_BASE, 'auth');
const userLoginUrl = urljoin(authUrl, 'login');
const userCredentialsUrl = urljoin(authUrl, 'me'); // GET / PATCH
const registerUrl = urljoin(authUrl, 'register'); // POST
const activateUrl = urljoin(authUrl, 'activate'); // POST
const usernameUrl = urljoin(authUrl, 'username'); // POST
const passwordUrl = urljoin(authUrl, 'password'); // POST
const resetPasswordUrl = urljoin(passwordUrl, 'reset'); // POST
const confirmPasswordUrl = urljoin(resetPasswordUrl, 'confirm'); // POST

//-----------------------------------------------------------------------------
//  USER
//-----------------------------------------------------------------------------
const userUrl = urljoin(API_BASE, 'user'); // GET all users (if admin, else 'me')
const userProfileUrl = urljoin(userUrl, 'me'); // GET user profile
const userSrsUrl = urljoin(userUrl, 'srs'); // POST to get review count
const userSyncUrl = urljoin(userUrl, 'sync'); // POST to sync with WK

//-----------------------------------------------------------------------------
//  REVIEWS
//-----------------------------------------------------------------------------
const reviewsUrl = urljoin(API_BASE, 'review'); // GET all ready reviews
const criticalReviewsUrl = urljoin(reviewsUrl, 'critical'); // GET critical
const currentReviewsUrl = urljoin(reviewsUrl, 'current'); // GET current review queue
const currentLessonsUrl = urljoin(reviewsUrl, 'lesson'); // GET current lesson queue
const reviewEntryUrl = (id) => urljoin(reviewsUrl, id); // GET single
const reviewCorrectUrl = (id) => urljoin(reviewEntryUrl(id), 'correct'); // POST correct answer
const reviewIncorrectUrl = (id) => urljoin(reviewEntryUrl(id), 'incorrect'); // POST incorrect answer
const lockReviewUrl = (id) => urljoin(reviewEntryUrl(id), 'hide'); // POST
const unlockReviewUrl = (id) => urljoin(reviewEntryUrl(id), 'unhide'); // POST

//-----------------------------------------------------------------------------
//  SYNONYMS
//-----------------------------------------------------------------------------
const synonymUrl = urljoin(API_BASE, 'synonym'); // POST add, GET get all
const synonymEntryUrl = (id) => urljoin(synonymUrl, id); // DELETE remove, GET one

//-----------------------------------------------------------------------------
//  VOCABULARY
//-----------------------------------------------------------------------------
const vocabularyUrl = urljoin(API_BASE, 'vocabulary'); // GET everything!
const vocabularyEntryUrl = (id) => urljoin(vocabularyUrl, id); // GET one
const readingUrl = urljoin(API_BASE, 'reading'); // GET all
const readingEntryUrl = (id) => urljoin(readingUrl, id); // GET one
const levelsUrl = urljoin(API_BASE, 'level'); // GET all
const levelEntryUrl = (id) => urljoin(levelsUrl, id); // GET one
const lockLevelUrl = (id) => urljoin(levelEntryUrl(id), 'lock'); // POST lock
const unlockLevelUrl = (id) => urljoin(levelEntryUrl(id), 'unlock'); // POST unlock

//-----------------------------------------------------------------------------
//  GENERAL
//-----------------------------------------------------------------------------
const faqUrl = urljoin(API_BASE, 'faq'); // GET all, POST create
const faqEntryUrl = (id) => urljoin(faqUrl, id); // GET, PUT update, PATCH partial update, DELETE
const announcementUrl = urljoin(API_BASE, 'announcement'); // GET all, POST create
const announcementEntryUrl = (id) => urljoin(announcementUrl, id); // GET one, PUT update, PATCH  partial update, DELETE
const contactUrl = urljoin(API_BASE, 'contact');

//-----------------------------------------------------------------------------
//  EXTERNAL
//-----------------------------------------------------------------------------
export const createJishoApiUrl = (keyword) => `//jisho.org/api/v1/search/words?keyword=${keyword}`;
export const createJishoUrl = (keyword) => `//jisho.org/search/${keyword}`;


//-----------------------------------------------------------------------------
//  AUTHORIZATION
//-----------------------------------------------------------------------------
export const getUserAuth = ({ id, username, email } = {}) => get(userCredentialsUrl, { id, username, email });
export const loginUser = ({ username, password } = {}) => post(userLoginUrl, { username, password });
export const updateUserAuth = ({ id, username, email } = {}) => patch(userCredentialsUrl, { id, username, email });
export const registerUser = ({ email, username, password, apiKey } = {}) =>
  post(registerUrl, { email, username, password, api_key: apiKey });

export const activateUser = ({ uid } = {}) => post(activateUrl, { uid });
export const changeUsername = ({ username } = {}) => post(usernameUrl, { username });
export const changePassword = ({ password } = {}) => post(passwordUrl, { password });
export const resetPassword = ({ email } = {}) => post(resetPasswordUrl, { email }); // send password reset email.
export const confirmPassword = () => post(confirmPasswordUrl); // finish reset password process

//-----------------------------------------------------------------------------
//  USER
//-----------------------------------------------------------------------------
export const getUsers = () => get(userUrl);
export const getUserProfile = () => get(userProfileUrl);
export const syncKw = () => post(userSrsUrl);
// true to force ALL users to sync with WK
export const syncWk = ({ fullSync = false } = {}) => post(userSyncUrl, { full_sync: fullSync });

//-----------------------------------------------------------------------------
//  REVIEWS
//-----------------------------------------------------------------------------
export const getReviews = ({
  id,
  offset,
  limit,
  meaningContains,
  srsLevel,
  srsLevelLt,
  srsLevelGt,
} = {}) => get(
  reviewsUrl,
  {
    level: id,
    offset,
    limit,
    meaning_contains: meaningContains,
    srs_level: srsLevel,
    srs_level_lt: srsLevelLt,
    srs_level_gt: srsLevelGt,
  }
);

export const getCriticalReviews = ({ offset, limit } = {}) => get(criticalReviewsUrl, { offset, limit });
export const getCurrentReviews = ({ offset, limit } = {}) => get(currentReviewsUrl, { offset, limit });
export const getCurrentLessons = ({ offset, limit } = {}) => get(currentLessonsUrl, { offset, limit });

export const getReviewEntry = ({ id }) => get(reviewEntryUrl(id));
export const recordReview = ({ id, isCorrect, previouslyIncorrect }) => {
  const body = {
    user_specific_id: id,
    user_correct: isCorrect,
    wrong_before: previouslyIncorrect,
  };
  return isCorrect ? post(reviewCorrectUrl(id), body) : post(reviewIncorrectUrl(id), body);
};

export const lockReview = ({ id }) => post(lockReviewUrl(id));
export const unlockReview = ({ id }) => post(unlockReviewUrl(id));
export const saveReviewNotes = ({ id, notes }) => patch(reviewEntryUrl(id), { notes });

//-----------------------------------------------------------------------------
//  SYNONYMS
//-----------------------------------------------------------------------------
export const addSynonym = ({ reviewId, character, kana } = {}) => post(
  synonymUrl,
  { review: reviewId, character, kana },
);
export const removeSynonym = ({ id }) => del(synonymEntryUrl(id));

//-----------------------------------------------------------------------------
//  VOCABULARY
//-----------------------------------------------------------------------------
export const getVocabulary = ({
  id: level,
  offset,
  limit,
  meaningContains,
  readingsKanaContains,
  readingsCharacterContains,
} = {}) => get(
  vocabularyUrl,
  {
    level,
    offset,
    limit,
    meaning_contains: meaningContains,
    readings_kana_contains: readingsKanaContains,
    readings_character_contains: readingsCharacterContains,
  }
);

export const getLevels = () => get(levelsUrl);

export const getVocabularyEntry = ({ id }) => get(vocabularyEntryUrl(id));

export const getReadings = ({ offset, limit } = {}) => get(readingUrl, { offset, limit });
export const getReadingEntry = ({ id }) => get(readingEntryUrl(id));

export const lockLevel = ({ id }) => post(lockLevelUrl(id));
export const unlockLevel = ({ id }) => post(unlockLevelUrl(id));

//-----------------------------------------------------------------------------
//  GENERAL
//-----------------------------------------------------------------------------
export const getFaqs = () => get(faqUrl);
export const getFaq = ({ id }) => get(faqEntryUrl(id));
export const addFaq = ({ question, answer } = {}) => post(faqUrl, { question, answer });
export const updateFaq = ({ question, answer } = {}) => put(faqUrl, { question, answer });

export const getAnnouncements = () => get(announcementUrl);
export const getAnnouncement = ({ id }) => get(announcementEntryUrl(id));
export const addAnnouncement = ({ title, body } = {}) => post(announcementUrl, { title, body });
export const updateAnnouncement = ({ title, body } = {}) => put(announcementUrl, { title, body });

export const sendContactMessage = ({ name, email, body } = {}) => post(contactUrl, { name, email, body });
