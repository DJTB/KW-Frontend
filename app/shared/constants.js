/**
 * Maps srs rank levels to english names for rank
 * @type {Object} 'ONE': 'apprentice'
 */
export const SRS_RANKS = {
  ONE: 'apprentice',
  TWO: 'guru',
  THREE: 'master',
  FOUR: 'enlightened',
  FIVE: 'burned',
};

/**
 * event.which keycodes
 * @type {Object} K_LOWERCASE: 75
 */
export const KEYS = {
  P_LOWERCASE: 80,
  P_UPPERCASE: 112,
  K_LOWERCASE: 75,
  K_UPPERCASE: 107,
  F_LOWERCASE: 70,
  F_UPPERCASE: 102,
  S_LOWERCASE: 83,
  S_UPPERCASE: 115,
  I_LOWERCASE: 73,
  I_UPPERCASE: 105,
  BACKSPACE: 8,
  FORWARD_SLASH: 191,
  BACK_SLASH: 220,
  ENTER: 13,
  DELETE: 46,
  TAB: 9,
  ESCAPE: 27,
  SPACE: 32,
  LEFT: 37,
  UP: 38,
  RIGHT: 39,
  DOWN: 40,
  PRIME: 222,
  COLON: 186,
  EQUALS: 187,
  COMMA: 188,
  HYPHEN: 189,
  PERIOD: 190,
  BACKTICK: 192,
  BRACKET_LEFT: 219,
  BRACKET_RIGHT: 221,
};
