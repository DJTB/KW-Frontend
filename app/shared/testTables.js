import { random } from 'lodash';

/**
 * These all assume that the data has been serialized to preferred format in the front end already!
 * This is not the API response shape.
 */

export const levels = [
  {
    level: 1,
    unlocked: false,
    count: 43,
  },
  {
    level: 2,
    unlocked: true,
    count: 90,
  },
  {
    level: 3,
    unlocked: true,
    count: 69,
  },
  {
    level: 4,
    unlocked: false,
    count: 105,
  },
  {
    level: 5,
    unlocked: true,
    count: 126,
  },
  {
    level: 6,
    unlocked: true,
    count: 118,
  },
  {
    level: 7,
    unlocked: true,
    count: 101,
  },
  {
    level: 8,
    unlocked: false,
    count: 136,
  },
  {
    level: 9,
    unlocked: false,
    count: 119,
  },
  {
    level: 10,
    unlocked: false,
    count: 118,
  },
  {
    level: 11,
    unlocked: false,
    count: 126,
  },
  {
    level: 12,
    unlocked: false,
    count: 126,
  },
  {
    level: 13,
    unlocked: false,
    count: 115,
  },
  {
    level: 14,
    unlocked: false,
    count: 116,
  },
  {
    level: 15,
    unlocked: false,
    count: 100,
  },
  {
    level: 16,
    unlocked: false,
    count: 117,
  },
  {
    level: 17,
    unlocked: false,
    count: 124,
  },
  {
    level: 18,
    unlocked: false,
    count: 135,
  },
  {
    level: 19,
    unlocked: false,
    count: 106,
  },
  {
    level: 20,
    unlocked: false,
    count: 114,
  },
  {
    level: 21,
    unlocked: false,
    count: 106,
  },
  {
    level: 22,
    unlocked: false,
    count: 115,
  },
  {
    level: 23,
    unlocked: false,
    count: 97,
  },
  {
    level: 24,
    unlocked: false,
    count: 120,
  },
  {
    level: 25,
    unlocked: false,
    count: 100,
  },
  {
    level: 26,
    unlocked: false,
    count: 124,
  },
  {
    level: 27,
    unlocked: false,
    count: 106,
  },
  {
    level: 28,
    unlocked: false,
    count: 116,
  },
  {
    level: 29,
    unlocked: false,
    count: 107,
  },
  {
    level: 30,
    unlocked: false,
    count: 103,
  },
  {
    level: 31,
    unlocked: false,
    count: 113,
  },
  {
    level: 32,
    unlocked: false,
    count: 111,
  },
  {
    level: 33,
    unlocked: false,
    count: 103,
  },
  {
    level: 34,
    unlocked: false,
    count: 124,
  },
  {
    level: 35,
    unlocked: false,
    count: 100,
  },
  {
    level: 36,
    unlocked: false,
    count: 98,
  },
  {
    level: 37,
    unlocked: false,
    count: 112,
  },
  {
    level: 38,
    unlocked: false,
    count: 100,
  },
  {
    level: 39,
    unlocked: false,
    count: 101,
  },
  {
    level: 40,
    unlocked: false,
    count: 116,
  },
  {
    level: 41,
    unlocked: false,
    count: 107,
  },
  {
    level: 42,
    unlocked: false,
    count: 96,
  },
  {
    level: 43,
    unlocked: false,
    count: 108,
  },
  {
    level: 44,
    unlocked: false,
    count: 103,
  },
  {
    level: 45,
    unlocked: false,
    count: 96,
  },
  {
    level: 46,
    unlocked: false,
    count: 103,
  },
  {
    level: 47,
    unlocked: false,
    count: 94,
  },
  {
    level: 48,
    unlocked: false,
    count: 104,
  },
  {
    level: 49,
    unlocked: false,
    count: 104,
  },
  {
    level: 50,
    unlocked: false,
    count: 87,
  },
  {
    level: 51,
    unlocked: false,
    count: 75,
  },
  {
    level: 52,
    unlocked: false,
    count: 99,
  },
  {
    level: 53,
    unlocked: false,
    count: 103,
  },
  {
    level: 54,
    unlocked: false,
    count: 112,
  },
  {
    level: 55,
    unlocked: false,
    count: 76,
  },
  {
    level: 56,
    unlocked: false,
    count: 88,
  },
  {
    level: 57,
    unlocked: false,
    count: 93,
  },
  {
    level: 58,
    unlocked: false,
    count: 81,
  },
  {
    level: 59,
    unlocked: false,
    count: 79,
  },
  {
    level: 60,
    unlocked: false,
    count: 65,
  },
];

export const vocabs = [
  {
    id: 553,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['before long', 'soon'],
    readings: [
      {
        id: 582,
        character: '近々',
        kana: ['ちかぢか'],
        level: 7,
        tags: ['Adverbial Noun'],
        sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
        sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
        jlpt: 'JLPT N2',
        common: true,
      },
      {
        id: 583,
        character: '近々',
        kana: ['きんきん'],
        level: 7,
        tags: ['Adverbial Noun'],
        sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
        sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
        jlpt: 'JLPT N2',
        common: true,
      },
      {
        id: 584,
        character: '近々',
        kana: ['ちかじか'],
        level: 7,
        tags: ['Adverbial Noun'],
        sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
        sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
        jlpt: 'JLPT N2',
        common: true,
      },
      {
        id: 783,
        character: '軈て',
        kana: ['やがて'],
        level: 61,
        tags: ['Adverb'],
        sentenceEn: 'By and by he forgot that affair',
        sentenceJa: 'やがて彼はその事件のことを忘れた',
        jlpt: 'JLPT N2',
        common: true,
      },
    ],
    synonyms: [
      {
        id: 5582,
        character: '間もなく',
        kana: ['まもなく'],
        level: 22,
        tags: ['Adverb'],
        sentenceEn: 'Coming up, Jiyuugaoka Station.',
        sentenceJa: '間もなく、自由が丘駅',
        jlpt: 'JLPT N2',
        common: true,
      },
    ],
  },
  {
    id: 176,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['young lady', 'little girl', 'maiden', 'daughter', 'girl'],
    readings: [
      {
        id: 188,
        character: '少女',
        kana: ['しょうじょ'],
        level: 3,
        tags: ['Noun'],
        sentenceEn: "I am touched by the girl's acute sensitivity.",
        sentenceJa: '少女の鋭い感性に触れている',
        jlpt: 'JLPT N3',
        common: true,
      },
      {
        id: 189,
        character: '少女',
        kana: ['おとめ'],
        level: 3,
        tags: ['Noun'],
        sentenceEn: "I am touched by the girl's acute sensitivity.",
        sentenceJa: '少女の鋭い感性に触れている',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
  {
    id: 280,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['young boy', 'juvenile', 'youth', 'boy'],
    readings: [
      {
        id: 298,
        character: '少年',
        kana: ['しょうねん'],
        level: 4,
        tags: ['Noun'],
        sentenceEn: 'The shy boy murmured his name.',
        sentenceJa: 'その内気な少年はぼそぼそと自分の名前をつぶやいた全ての少年少女はを知っています',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
  {
    id: 424,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: [
      'whole life',
      'entire life',
      'a lifetime',
      'all through life',
      'one existence',
      'a generation',
      'an age',
    ],
    readings: [
      {
        id: 449,
        character: '一生',
        kana: ['いっしょう'],
        level: 5,
        tags: ['Noun', 'No adjective'],
        sentenceEn: 'She worked hard.',
        sentenceJa: '彼女は一生懸命働いた',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
  {
    id: 502,
    history: {
      correct: random(10),
      incorrect: random(10),
    },
    session: {
      correct: random(10),
      incorrect: random(10),
      streak: random(11),
    },
    meanings: ['honest'],
    readings: [
      {
        id: 529,
        character: '正直',
        kana: ['しょうじき'],
        level: 6,
        tags: ['Noun', 'Na adjective'],
        sentenceEn: 'Him, honest? What a joke!',
        sentenceJa: '正直が聞いて呆れるよ',
        jlpt: 'JLPT N3',
        common: true,
      },
    ],
    synonyms: [],
  },
];

export const readings = {
  single: [
    {
      id: 351,
      character: '売り上げ',
      kana: ['うりあげ'],
      level: 5,
      tags: ['Noun'],
      sentenceEn: 'Sales are down this month.',
      sentenceJa: '今月は売上が落ちた',
      jlpt: 'JLPT N2',
      common: true,
    },
  ],
  sameCharacters: [
    {
      id: 582,
      character: '近々',
      kana: ['ちかぢか'],
      level: 7,
      tags: ['Noun'],
      sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
      sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
      jlpt: 'JLPT N2',
      common: true,
    },
    {
      id: 583,
      character: '近々',
      kana: ['きんきん'],
      level: 7,
      tags: ['Noun'],
      furigana: '0:きんきん',
      sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
      sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
      jlpt: 'JLPT N2',
      common: true,
    },
    {
      id: 584,
      character: '近々',
      kana: ['ちかじか'],
      level: 7,
      tags: ['Noun'],
      sentenceEn: 'When I saw them on a date, I asked them, “Do I hear wedding bells?”',
      sentenceJa: '彼らがデートをしているのを見かけて近々結婚するのってきいたんだ',
      jlpt: 'JLPT N2',
      common: true,
    },
  ],
  suffix: [
    {
      id: 2476,
      character: '各〜',
      kana: ['かく'],
      level: 22,
      tags: ['Prefix'],
      sentenceEn: 'First, choose a representative for each team.',
      sentenceJa: 'まず始めに、各チームの代表者を決めて下さい',
      jlpt: null,
      common: null,
    },
  ],
  prefix: [
    {
      id: 6337,
      character: '〜漬け',
      kana: ['づけ'],
      level: 59,
      tags: ['Suffix'],
      sentenceEn: 'These are lightly pickled cucumbers.',
      sentenceJa: '胡瓜の浅漬けならありますよ',
      jlpt: null,
      common: null,
    },
    {
      id: 130,
      character: '〜才',
      kana: ['さい'],
      level: 2,
      tags: ['Suffix'],
      sentenceEn: 'I am five years old.',
      sentenceJa: 'わたしは５才です',
      jlpt: null,
      common: null,
    },
  ],
  mixed: [
    {
      id: 933,
      character: 'ハート形',
      kana: ['ハートがた'],
      level: 9,
      tags: ['Noun'],
      sentenceEn: 'When flamingos kiss, their heads and necks form a heart shape.',
      sentenceJa: 'フラミンゴがキスをする時、二羽の頭と首がハート形を描く',
      jlpt: null,
      common: false,
    },
    {
      id: 30,
      character: 'フランス人',
      kana: ['ふらんすじん'],
      level: 1,
      tags: ['Noun'],
      sentenceEn: 'He is French.',
      sentenceJa: '彼はフランス人です',
      jlpt: null,
      common: false,
    },
    {
      id: 3190,
      character: '迷惑メール',
      kana: ['めいわくメール'],
      level: 29,
      tags: ['Noun'],
      sentenceEn:
        'He sent about 2.2 billion spam emails over a period of around one and a half years.',
      sentenceJa: '約１年半で約億通の迷惑メールを送りました',
      jlpt: null,
      common: false,
    },
    {
      id: 6193,
      character: '烏龍茶',
      kana: ['ウーロンちゃ'],
      level: 57,
      tags: ['Noun'],
      sentenceEn: 'Oolong tea without ice, please.',
      sentenceJa: '烏龍茶を氷抜きでください',
      jlpt: null,
      common: null,
    },
    {
      id: 4875,
      character: '缶ビール',
      kana: ['かんビール'],
      level: 44,
      tags: ['Noun'],
      sentenceEn: 'Oh, this canned beer is way too good for me.',
      sentenceJa: 'ああ、この缶ビールはマジでうますぎる',
      jlpt: null,
      common: false,
    },
  ],
  onlyKana: [
    {
      id: 9000,
      common: true,
      jlpt: 'JLPT N4',
      character: 'そろそろ',
      kana: ['そろそろ'],
      tags: ['Adverb', "Adverb taking the 'to' particle"],
      sentenceJa: '「あの、かりんさん...、そろそろお開きにしませんか？」「駄目、勝ち逃げ禁止！今度こそボクが勝つの！」',
      sentenceEn:
        "“Er, Karin ..., shouldn't we call it a night soon?” “No! No quitting while ahead! Next time I'll win for sure!”",
    },
    {
      id: 9001,
      common: false,
      jlpt: null,
      character: 'マンコ',
      kana: ['マンコ'],
      tags: ['Noun'],
      sentenceJa: 'マンコいいな〜',
      sentenceEn: 'What a great pussy.',
    },
  ],
  partialKanaSentence: [
    {
      id: 8800,
      common: true,
      jlpt: 'JLPT N4',
      character: '付く',
      kana: ['つく'],
      tags: ['Transitive Verb'],
      sentenceJa: 'この「オンボード」というのは平たく言えば「グラフィックカードがついていない」ものだ',
      sentenceEn:
        'Frankly speaking, this “on board” means “doesn’t have a graphics card”.',
    },
  ],
};
