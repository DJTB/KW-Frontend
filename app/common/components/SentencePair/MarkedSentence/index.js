import React from 'react';
import PropTypes from 'prop-types';
import splitSentenceByMatch from 'common/utils/splitSentenceByMatch';
import A from 'common/components/A';

import { Sentence, VocabMark } from './styles';

MarkedSentence.propTypes = {
  sentence: PropTypes.string.isRequired,
  word: PropTypes.string,
  reading: PropTypes.string,
  verbType: PropTypes.string,
};

MarkedSentence.defaultProps = {
  word: '',
  reading: '',
  verbType: '',
};

function MarkedSentence({ sentence, word, reading, verbType }) {
  const { head, match, tail } = splitSentenceByMatch({ sentence, word, reading, verbType });

  return (
    <Sentence lang="ja">
      <A
        href={`http://jisho.org/search/${sentence}`}
        title="Deconstruct sentence at Jisho.org"
        plainLink
        external
      >
        <span>{head}</span>
        <VocabMark>{match}</VocabMark>
        <span>{tail}</span>
      </A>
    </Sentence>
  );
}

export default MarkedSentence;
