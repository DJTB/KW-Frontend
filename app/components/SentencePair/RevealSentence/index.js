import React from "react";
import PropTypes from "prop-types";

import { Sentence, RevealIcon } from "./styles";

RevealSentence.propTypes = {
  sentence: PropTypes.string.isRequired,
};

function RevealSentence({ sentence }) {
  return (
    <div>
      <Sentence tabIndex="0">
        {sentence}
        <RevealIcon name="EYE" />
      </Sentence>
    </div>
  );
}

export default RevealSentence;
