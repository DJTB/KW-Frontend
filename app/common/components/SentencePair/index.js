import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import {
  selectWord,
  selectPrimaryReading,
  selectSentenceEn,
  selectSentenceJa,
  selectVerbType,
} from 'features/vocab/selectors';

import P from 'common/components/P';
import MarkedSentence from './MarkedSentence';
import RevealSentence from './RevealSentence';
import { Wrapper } from './styles';

SentencePair.propTypes = {
  word: PropTypes.string.isRequired,
  primaryReading: PropTypes.string.isRequired,
  sentenceEn: PropTypes.string,
  sentenceJa: PropTypes.string,
  verbType: PropTypes.string,
};

SentencePair.defaultProps = {
  sentenceEn: '',
  sentenceJa: '',
  verbType: '',
};

export function SentencePair({ word, primaryReading, sentenceEn, sentenceJa, verbType }) {
  const hasData = sentenceEn && sentenceJa;
  return hasData ? (
    <Wrapper>
      <MarkedSentence
        sentence={sentenceJa}
        word={word}
        reading={primaryReading}
        verbType={verbType}
      />
      <RevealSentence sentence={sentenceEn} />
    </Wrapper>
  ) : (
    <Wrapper>
      <P>No sentence data found.</P>
    </Wrapper>
  );
}

const mapStateToProps = (state, props) => ({
  word: selectWord(state, props),
  primaryReading: selectPrimaryReading(state, props),
  sentenceEn: selectSentenceEn(state, props),
  sentenceJa: selectSentenceJa(state, props),
  verbType: selectVerbType(state, props),
});

export default connect(mapStateToProps)(SentencePair);
