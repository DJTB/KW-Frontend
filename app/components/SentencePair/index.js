import React from "react";
import PropTypes from "prop-types";

import MarkedSentence from "./MarkedSentence";
import RevealSentence from "./RevealSentence";
import { Wrapper } from "./styles";

SentencePair.propTypes = {
  sentenceEn: PropTypes.string.isRequired,
  sentenceJa: PropTypes.string.isRequired,
  character: PropTypes.string.isRequired,
  kana: PropTypes.array.isRequired,
};

// TODO: placeholder explaining no sentence data - better than blowing up

function SentencePair({
  sentenceEn, sentenceJa, character, kana,
}) {
  return (
    <Wrapper>
      <MarkedSentence sentence={sentenceJa} character={character} kana={kana} />
      <RevealSentence sentence={sentenceEn} />
    </Wrapper>
  );
}

export default SentencePair;
