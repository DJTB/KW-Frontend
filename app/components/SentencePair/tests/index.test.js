import 'jest-styled-components';
import React from 'react';
import { render } from 'enzyme';

import { readings } from 'shared/testTables';
import SentencePair from '../index';

describe('<SentencePair />', () => {
  it('should match baseline snapshot', () => {
    const renderedComponent = render(<SentencePair {...readings.single[0]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
