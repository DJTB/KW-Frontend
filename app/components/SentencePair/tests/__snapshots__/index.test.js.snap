// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SentencePair /> should match baseline snapshot 1`] = `
<div>
  <p
    class="sMpVQ iFxjYr"
    lang="ja"
  >
    <a
      class="dTJZJM"
      href="http://jisho.org/search/その言葉の漢字は難しい"
      rel="external noopener noreferrer"
      target="_blank"
      title="Deconstruct sentence at Jisho.org"
    >
      <span>
        その言葉の
      </span>
      <mark
        class="hMFuqV cjlhSZ"
        color="inherit"
      >
        漢字
      </mark>
      <span>
        は難しい
      </span>
    </a>
  </p>
  <p
    class="sMpVQ iFxjYr"
  >
    This sentence's phrase is difficult
  </p>
</div>
`;
